audio-worklet/src/noisegate-audio-worklet.js

```diff
@@ -0,0 +1,216 @@
+    // to enable smooth transitions between sound and silence.
+    // The default value of 50ms has been set experimentally to minimize
+    // glitches in the demo.
+    {
+      name: 'attack',
+      defaultValue: 0.05,
+      minValue: 0,
+      maxValue: 0.1
+    },
+    {
+      name: 'release',
+      defaultValue: 0.05,
+      minValue: 0,
+      maxValue: 0.1
+    },
+    // The maximum threshold is 0 since that is the defined maximum in dBFS.
+    // The minimum is -100 dBFS since the sound is inaudible at that
+    // level even without the noise gate's interference. The default is set to
+    // -40 since the noise in the demo (with a gain of 1) will be muted during
+    // pauses at this value but the recorded voice sample will not be muted.
+    {
+      name: 'threshold',
+      defaultValue: -40,
+      minValue: -100,
+      maxValue: 0
+    },
+    // The default timeConstant has been set experimentally to 0.0025s to
+    // balance delay for high frequency suppression. The maximum value is set
+    // somewhat arbitrarily at 0.1 since the envelope is very delayed at values
+    // beyond this.
+    {
+      name: 'timeConstant',
+      defaultValue: 0.0025,
+      minValue: 0,
+      maxValue: 0.1
+    }];
+  }
+
+  constructor() {
+    super();
+
+    // The previous envelope level, a float representing signal amplitude.
+    this.previousLevel_ = 0;
+
+    // The last weight (between 0 and 1) assigned, where 1 means the gate
+    // is open and 0 means it is closed and the sample in the output buffer is
+    // muted. When attacking, the weight will linearly decrease from 1 to 0, and
+    // when releasing the weight linearly increase from 0 to 1.
+    this.previousWeight_ = 1.0;
+    this.envelope_ = new Float32Array(128);
+    this.weights_ = new Float32Array(128);
+
+    // TODO (issue #111): Use getContextInfo() to get sample rate.
+    this.sampleRate = 44100;
+  }
+
+  /**
+   * Control the dynamic range of input based on specified threshold.
+   * @param {AudioBuffer} input Input audio data
+   * @param {AudioBuffer} output Output audio data
+   * @param {Object} parameters K-rate audio params.
+   * @param {Number} parameters.attack Seconds for gate to fully close.
+   * @param {Number} parameters.release Seconds for gate to fully open.
+   * @param {Number} parameters.timeConstant Seconds for envelope follower's
+   *                                         smoothing filter delay.
+   * @param {Number} parameters.threshold Decibel level beneath which sound is
+   *                                      muted.
+   */
+  process(input, output, parameters) {
+    // Alpha controls a tradeoff between the smoothness of the
+    // envelope and its delay, with a higher value giving more smoothness at
+    // the expense of delay and vice versa.
+    this.alpha_ = this.getAlphaFromTimeConstant_(
+        parameters.timeConstant[0], this.sampleRate);
+
+    // The a-rate audio-params are used as k-rate audio-params.
```

Comment:
Perhaps

`// k-rate: use the first element of parameter data to process each render quantum.`